src/signalrclient/hub_connection_impl.cpp

```diff
@@ -70,7 +70,8 @@ namespace signalr
             return iter->second;
         }
 
-        auto proxy = std::make_shared<internal_hub_proxy>(hub_name, m_logger);
+        auto weak_connection = std::weak_ptr<hub_connection_impl>(shared_from_this());

```

Comment:
Is the weak_ptr here to prevent this method from blocking the connection from getting destroyed?